### PR TITLE
Extract histograms and bars into RzCons

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- Extract histograms code into `librz/cons/histogram.c`
- Extract bars (progressbar and rangebar) into `librz/cons/bar.c`
- Initially I wanted to put them into `librz/util/` but they use RzConsPalette for colors, so I put them into RzCons

**Test plan**

- CI is green
- Play with various `p=` and `p-` commands

**Closing issues**

A prerequisite for converting `p-` and `p=` to rzhell: https://github.com/rizinorg/rizin/issues/1590

### CODE REVIEW

File: librz/cons/bar.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+// TODO: add support for colors
+/**
+ * \brief Create the string buffer with the progressbar
+ *
+ * \param opts Progressbar options: color, style, legend
+ * \param pc How much percent is filled
+ * \param width Width of the histogram
+ */
+RZ_API RZ_OWN RzStrBuf *rz_progressbar(RZ_NONNULL RzBarOptions *opts, int pc, int width) {
+	rz_return_val_if_fail(opts, NULL);
```
Comment (wargio) [inline librz/cons/bar.c:16]: forbid negative maybe? or enforce unsigned

File: librz/cons/bar.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	pc = RZ_MAX(0, RZ_MIN(100, pc));
+	if (opts->legend) {
+		rz_strbuf_appendf(buf, "%4d%% ", pc);
+	}
+	cols -= 15;
+	rz_strbuf_append(buf, "[");
+	for (i = cols * pc / 100; i; i--) {
+		rz_strbuf_append(buf, block);
+	}
+	for (i = cols - (cols * pc / 100); i; i--) {
```
Comment (wargio) [inline librz/cons/bar.c:34]: `i` can be defined in the loop.

**Review by wargio — APPROVED** [review]
LGTM, besides a couple of comments

File: librz/cons/bar.c (hunk 2 above)
Comment (XVilka) [reply librz/cons/bar.c:34]: It's reused in two different loops, I don't think it's really beneficial.

Comment (wargio) [reply librz/cons/bar.c:34]: yes, but you can redefine it so it gets optimized by the compiler.

**Review by wargio — APPROVED** [review]

**Pull request merged**